src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +427,69 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
```

Comment:
I don't think this issue is fixed. In the above example, `$otherForeignKeys` would be `['site_id', 'role_id']` after all. But the expected target foreign key is only `site_id`.